vyper/semantics/validation/annotation.py

```diff
@@ -188,7 +210,8 @@ def visit_List(self, node, type_):
             self.visit(element, type_.value_type)
 
     def visit_Name(self, node, type_):
-        node._metadata["type"] = get_exact_type_from_node(node)
+        if "type" not in node._metadata:
```

Comment:
when could this be annotated already? are we sure we don't want to overwrite it / are we sure `node._metadata["type"]` is better than `type_`?

Reply:
When the `Name` node is a typestring for builtin functions, it is first annotated with a `TypeTypeDefinition`. Without this check, it will be overwritten by the `BaseTypeDefinition` instead. This results in an issue when it is then handled in `process_arg` in `vyper/builtin_functions/signatures.py`.

Reply:
hm, why not use `type_` like the other visitor functions? i suspect that would fix it

Reply:
I think this check is still required. For example, immutable `Name` nodes are annotated in `vyper/semantics/validation/module.py` and removing this check will throw an error because the immutable identifier is not in the namespace at this stage.